runtime/near-vm-logic/src/alt_bn128.rs

```diff
@@ -1,348 +1,169 @@
-use bn::arith::U256;
-use bn::{pairing_batch, AffineG1, AffineG2, Fq, Fq2, Fr, Group, GroupError, Gt, G1, G2};
-use borsh::{BorshDeserialize, BorshSerialize};
-use near_vm_errors::VMLogicError;
-use std::io::{self, Error, ErrorKind, Write};
+use crate::array_utils::{join_array, split_array, ArrayChunks};
+use bn::Group;
+use near_vm_errors::{HostError, VMLogicError};
 
-use crate::HostError;
+const BOOL_SIZE: usize = 1;
+const U128_SIZE: usize = 128 / 8;
```

Comment:
`u128::BITS` feels semantically wrong here -- this essentially describes serialized representation which is independent of any particular PL, not the property of Rust programming language. hence, bool size is `1`, rather than `size_of::<bool>`. But yeah, inlining this is a good call!

Reply:
Well, the serialization implementation relies on `u128::from_le_bytes` which ultimately ties the entire thing back into specifics of how Rust implements the type.